FantasmoSDK/Classes/FMLocationManager.swift

```diff
@@ -265,11 +277,10 @@ open class FMLocationManager: NSObject, FMApiDelegate {
 extension FMLocationManager : ARSessionDelegate {
     public func session(_ session: ARSession, didUpdate frame: ARFrame) {
         lastFrame = frame
-                        
-        guard state == .localizing && qualityFilter.accepts(frame)
-        else { return }
         
-        localize(frame: frame)
+        if state == .localizing && qualityFilter.accepts(frame) {
```

Comment:
Why not use `guard`? It seems semantically appropriate here.

Reply:
Will try to explain in the following way. If we had the following implementation of function:

```public func session(_ session: ARSession, didUpdate frame: ARFrame) {
        lastFrame = frame
        
        // Some code
        // Some code
        // Some code
        // Some code
        // Some code
        // Some code
        // Some code

        guard state == .localizing && qualityFilter.accepts(frame) else {
            return
        }
        
        localize(frame: frame, from: session)
    }```
    
    We would prefer `if` to `guard`.  
    
    If this guard referred to the whole function body its usage would be totally ok from my point of view, or if piece of code after guard spread significant number of lines then also usage would be justified, though even in this latter case I would prefer `if`. When func becomes big early return hampers its debugging. This is why I don't like these guards actually. Good code have small functions, ideally not more than 20 lines, and using guards is justified. Here function is small and early return can be used without fear, but from my point of view only if it locates at the very beginning of the func. I prefer always using `if`, and only if `guard` is noticeably more appropriate I use it. `if` tends to visually couple conditions with corresponding code stronger than `guard`. But when there is a lot of nested `if`s it is better to resort to early return (or `guard`).
    
I think `guard` initially was introduced to avoid "pyramids of doom" as they call it, that is to avoid multiple levels of nesting of `if` when working with conditional types and I think this is its main purpose - to perform conditional binding. For other cases, I think, it's better to go for `if`.

And, at the same time I want to give in.  I have reverted to the earlier version.

Reply:
And here we have side effect related with quality filter. This filter is with memory and frame can change its state. It is good idea to express this fact somehow in the name of function of somehow else. This is why I wanted to call it "sequence filter", that is kind of hint, that filter anticipates not just occasional frames at the entrance but frames from some sequence. This is slightly suggests that filter has memory.

Reply:
I have suggestion. To rename `FMFrameFilter.accepts` to `FRFrameFilter.acceptAndCheck`. "Accept" part hints at side effect and "check" part convey the main purpose of the func. Such a name is more aligned with https://take.ms/b8Um8.